### PR TITLE
Add no-viewer mode and tests

### PR DESCRIPTION
**Pull request by pFernbach**
* Minor changes in the classes running the guide planning and the contact generation to be able to run with or without a viewer
* Add an argument --no-viewer to the main script to disable the use of gepetto-gui
* Add python tests that run the guide planning and the contact generation for Talos walking on flat floor
* Add python unit tests for State and StateHelper classes

Solve https://github.com/humanoid-path-planner/hpp-rbprm-corba/issues/60

# Other changes:
* Small fixes in the Scenarios classes:
    * interpolate correctly use the specified dt
    * By default, the humanoid robot only add the legs as Limb
    * When changing the name of the urdf, correctly specify the full name and update the visual name
* Add a demo on the floor of Bauzil with small obstacles on the ground
* Fix compilation warnings

### CODE REVIEW

**Comment by jmirabel** [comment]
This comes probably a bit too late but the information may be useful in the future. I just proposed to add a fake client into gepetto-viewer-corba so that, if it is not possible to connect to the viewer, then you get an object with the same API as the viewer.
See https://github.com/Gepetto/gepetto-viewer-corba/pull/140

File: src/hpp/corbaserver/rbprm/__main__.py [hunk 1]
```diff
@@ -31,27 +33,32 @@
...
 process_server = subprocess.Popen("hpp-rbprm-server",
                                   stdout=subprocess.PIPE,
                                   stderr=subprocess.DEVNULL,
                                   preexec_fn=os.setpgrp)
+# register cleanup methods to kill server when exiting python interpreter
+atexit.register(process_server.kill)
+
+# do the same for the viewer, exept if --no-viewer flag is set
+disable_viewer = args.no_viewer
+if disable_viewer is None:
```
Comment (nim65s) [inline src/hpp/corbaserver/rbprm/__main__.py]: `action="store_true"` shoud set `args.no_viewer` to either True or False, so I suggest:
`if not args.no_viewer:`

**Review by nim65s — APPROVED** [review]
Everything looks good to me, apart from a tiny argparse use. But maybe we can consider waiting for @jmirabel 's PR to avoid adding some of the logic here.

**Comment by pFernbach** [comment]
Ok, I also think that it's better to wait the update in gepetto-viewer and use the same thing everywhere.

**Comment by jmirabel** [comment]
This isn't used anywhere and hasn't been tested yet. If those changes are urgent, then don't wait :)

**Comment by pFernbach** [comment]
It's not urgent, but it should be in the next release. Like the changes in gepetto-viewer I guess.

**Comment by pFernbach** [comment]
Done, it require https://github.com/humanoid-path-planner/hpp-gepetto-viewer/pull/36 and https://github.com/humanoid-path-planner/hpp-affordance-corba/pull/18

**Pull request merged**